solutions/remotemonitoring/armtemplates/basic-static-map.json

```diff
@@ -407,6 +413,63 @@
                 "description": "Diagnostics service endpoint url"
             },
             "defaultValue": "Undefined"
+        },
+        "telemetryStorageType": {
+            "type": "string",
+            "defaultValue": "tsi",
+            "allowedValues": [
+                "tsi", "cosmosdb"
+            ],
+            "metadata": {
+                "description": "Storage type for telemetry messages, support TSI and CosmosDb"
+            }
+        },
+        "tsiEnvironmentName": {
+            "type": "string",
+            "defaultValue": "[concat('tsi-', take(uniqueString(subscription().subscriptionId, resourceGroup().id, parameters('solutionName')), 5))]",
```

Comment:
What is the 5 for?

Reply:
this is a length of random string by default, just like the other resource naming such as cosmos db, IoTHub.